### PR TITLE
RDoc-655 implementation

### PR DESCRIPTION
**Pull request by myarichuk**

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown [hunk 1]
```diff
@@ -0,0 +1,90 @@
+﻿# Document conflict in client-side
+  
+## What are conflicts?
+When two or more changes of a single document are done concurrently in two separate nodes, RavenDB cannot know which one of the changes is the correct one. This is called document conflict.  
+{NOTE For more information about conflicts and their resolution, see [article about conflicts](../../server/clustering/replication-conflicts). /}
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: Explicitly note here that this is NOT the default behavior.

File: Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+  
+## When is conflict except thrown?
+DocumentConflictException will be thrown _only_ if there is a fetch of document by id 
+and that document is conflicted. Otherwise conflicted documents are ignored. 
+This means during "starts with" queries, streaming, regular queries etc. the conflicted
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: Are you sure that this is the case in 4.0?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+documents are not retrieved and no exceptions are thrown.
+
+### Additionally
+ Fetching attachments of a conflicted document will throw `InvalidOperationException` on the server.
+
+## How the conflict can be resolved from the client side?
+ * PUT of a document with id that belongs to conflicted document will resolve the conflict.
+
+{CODE-BLOCK:csharp}
+using (var session = store.OpenSession())
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: Please don't embed code the code directly in markdown files. Put it in `Raven.Documentation.Samples` project:
https://github.com/ravendb/docs/tree/master/Documentation/4.0/Samples/csharp/Raven.Documentation.Samples

and use {CODE /} for that. e.g.:

{CODE document_store_creation@ClientApi\CreatingDocumentStore.cs /}


BTW Typically in Client API section we show Sync / Async examples. However it's not important for this article so we can show only sync examples here.

File: Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                    return final; //the conflict will be resolved to this variant
+                    "
+            }
+        }
+    };
+
+    var op = new ModifyConflictSolverOperation(
+        store.Database,
+        resolveByCollection, //we specify conflict resolution scripts by document collection 
+        resolveToLatest: false); //if true, RavenDB will resolve conflict to the latest.
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: WRT "will resolve conflict to the latest." - even if the script is specified? I believe it's not the case but the comments is confusing

Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: No need to adjust this comment? I _suppose_ it works as follow: _"if true, RavenDB will resolve conflict to the latest if there is no resolver defined for a given collection or the script returns `null`"_ but please confirm

File: Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown [hunk 5]
```diff
@@ -0,0 +1,29 @@
+﻿# Document conflict in client-side
+  
+## What are conflicts?
+When two or more changes of a single document are done concurrently in two separate nodes, 
+RavenDB cannot know which one of the changes is the correct one. This is called document conflict.  
+For more information about conflicts and their resolution, see [article about conflicts](../../server/clustering/replication-conflicts).  
+
+{NOTE By default, RavenDB will solve conflicts using "resolve to latest" strategy, thus the conflict will be resolved to a document with the latest 'modified date'./}
```
Comment (arekpalinski) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/cluster/document-conflicts-in-client-side.markdown]: What about possible time shifts between machines? I suppose the time synchronization is required but I see nothing mentioned about that in https://ravendb.net/docs/article-page/4.0/csharp/server/clustering/replication/replication-conflicts, should we do it?

**Pull request merged**